src/main/java/com/contrastsecurity/sdk/ContrastSDK.java

```diff
@@ -1830,13 +1831,29 @@ public HttpURLConnection makeConnection(String url, String method) throws IOExce
       connection.setRequestProperty(RequestConstants.TELEMETRY_INTEGRATION_VERSION, version);
     }
 
+    connection.setRequestProperty(HttpHeaders.USER_AGENT, buildUserAgentString().toString());
+
     connection.setUseCaches(false);
     if (connectionTimeout > DEFAULT_CONNECTION_TIMEOUT)
       connection.setConnectTimeout(connectionTimeout);
     if (readTimeout > DEFAULT_READ_TIMEOUT) connection.setReadTimeout(readTimeout);
     return connection;
   }
 
+  private StringBuilder buildUserAgentString() {
+    final StringBuilder userAgent = new StringBuilder();
+
+    if (integrationName != null) {
```

Comment:
Can we rename `integrationName` and `integrationVersion` to `componentName` and `componentVersion`?

"integration" is something that only Contrast cares about, but this SDK is a general library that users can build things with too. "Component" comes from the language describing the User-Agent header. See https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/User-Agent.

Reply:
I used the word "product" instead of "component", because that is consistent with the HTTP specification.